qubekit/torsions/fitting/forcebalance_wrapper.py

```diff
@@ -324,16 +325,31 @@ def _optimise(self, molecule: Ligand) -> Ligand:
             # back to fitting folder
             os.chdir(fitting_folder)
             # now we can make the optimize in file
-            self.generate_optimise_in(target_data=fitting_targets)
+            wq_port = self.generate_optimise_in(target_data=fitting_targets)
             # now make the forcefield file
             self.generate_forcefield(molecule=molecule)
 
             # now execute forcebalance
             with open("log.txt", "w") as log:
-                subprocess.run(
-                    "ForceBalance optimize.in", shell=True, stdout=log, stderr=log
+                forcebalance_process = subprocess.Popen(
+                    ["ForceBalance", "optimize.in"], stdout=log, stderr=log
                 )
 
+                import work_queue
+
+                workers = work_queue.Factory(
+                    "local", manager_host_port=f"localhost:{wq_port}"
+                )
+                # the optimisation takes space in gas, so 1 core for each OpenMM is good enough most likely
+                workers.cores = 1
+                # divide the memory for each worker
+                workers.memory = local_options.memory / local_options[0].cores
+                workers.min_workers = 1
+                workers.max_workers = local_options.cores
+
+                with workers:
+                    forcebalance_process.wait()
```

Comment:
Is there a reason this needs to change to `Popen` ? would the above work if we keep the old layout like 
```python
import work_queue
workers = work_queue.Factory(
                    "local", manager_host_port=f"localhost:{wq_port}"
                )
# the optimisation takes space in gas, so 1 core for each OpenMM is good enough most likely
workers.cores = 1
# divide the memory for each worker
workers.memory = local_options.memory / local_options[0].cores
workers.min_workers = 1
workers.max_workers = local_options.cores

with workers:
    subprocess.run("ForceBalance optimize.in", shell=True, stdout=log, stderr=log)
```

Reply:
ahh yes the order here I guess should not matter, this should work fine!